fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java

```diff
@@ -253,4 +260,37 @@ private String getResourcePath(final PersistentStorageSession pSession, final St
     private List<Link> getLinkHeaders(final List<String> headers) {
         return headers == null ? null : headers.stream().map(p -> Link.valueOf(p)).collect(Collectors.toList());
     }
+
+    /**
+     * Add this pairing to the containment index.
+     * @param txId The transaction ID or null if no transaction.
+     * @param parentId The parent ID or null if repository root.
+     * @param id The child ID.
+     */
+    private void addToContainmentIndex(final String txId, final String parentId, final String id) {
+        containmentIndex.addContainedBy(txId, ensurePrefix(parentId), ensurePrefix(id));
+    }
+
+    /**
+     * If the Slug has a forward slash then check that part of it is not an existing resource.
+     * @param txID The current transaction id.
+     * @param parent The fedora ID passed in with the slug.
+     * @param slug The slug.
+     * @return Array with the parent and slug.
+     */
+    private String[] splitSlug(final String txID, final String parent, final String slug) {
```

Comment:
I agree that this is somewhat similar except that a Slug could be the full path or just part of one, but is not an ID.

I'm not a fan of the current compound slug behaviour (ie. the following are equivalent)
```
POST http://localhost:8080/rest/A Slug: B
POST http://localhost:8080/rest Slug: A/B
```
but I was trying to avoid changing that behaviour. I can try defering some of this to the existing findExistingAncestor method.

Reply:
This is a niave question, but, outside of the HTTP layer, is there a reason that the code shouldn't interact with those two ids the same? That is to say regardless of the format of the request, the HTTP layer produces a consistent internal fedora id and the rest of the service doesn't need to care about if the slug was `B` or `A/B`?

Reply:
Basically because we need access to the persistence layer to check for object existence and that is in the kernel level. So we have to pass the information down. I guess we could stick the Slug on the end of the fedoraId.

My concern is in a deep Slug (ie. B/C/D) POSTed to A, would I change the fedoraId to A/B/C and the Slug to D? Because technically if B and C don't exist I need to create them too. Maybe not a huge issue, but something to consider.

Reply:
@pwinckles I refactored this function and how it is called a bit, see if this makes it better.